pkg/scheduler/scheduler.go

```diff
@@ -516,10 +516,13 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 
 	// Run the Reserve method of reserve plugins.
 	if sts := prof.RunReservePluginsReserve(schedulingCycleCtx, state, assumedPod, scheduleResult.SuggestedHost); !sts.IsSuccess() {
-		sched.recordSchedulingFailure(prof, assumedPodInfo, sts.AsError(), SchedulerError, "")
 		metrics.PodScheduleError(prof.Name, metrics.SinceInSeconds(start))
 		// trigger un-reserve to clean up state associated with the reserved Pod
 		prof.RunReservePluginsUnreserve(schedulingCycleCtx, state, assumedPod, scheduleResult.SuggestedHost)
+		if forgetErr := sched.Cache().ForgetPod(assumedPod); forgetErr != nil {
+			klog.Errorf("scheduler cache ForgetPod failed: %v", forgetErr)
+		}
+		sched.recordSchedulingFailure(prof, assumedPodInfo, sts.AsError(), SchedulerError, "")
```

Comment:
- put `recordSchedulingFailure` after the `Forget` because if the re-enqueuing & pop-out of that failed Pod happens super fast, the pod update event may still be skipped (see https://github.com/kubernetes/kubernetes/pull/93511#issuecomment-665316132)
- put `ForgetPod` after `RunReservePluginsUnreserve` for consistency (assume -> reserve -> unreserve -> unassume)